src/pyuvdata/uvdata/uvdata.py

```diff
@@ -5701,6 +5687,9 @@ def __add__(
                 raise ValueError(msg)
 
         # Begin manipulating the objects.
+        # Note that this will check to see if we can merge the telescopes (if they are
+        # different, otherwise the underlying checking is the same).
+        this.telescope += other.telescope
```

Comment:
What happens if the telescopes are the same (probably the most common situation)? Is this a no-op?